### PR TITLE
feat(react-core/card): added Card Components

### PR DESCRIPTION
**Pull request by dmiller9911**
affects: @patternfly/react-core, @patternfly/react-docs

Added Card, CardHeader, CardBody, and CardFooter components

ISSUES CLOSED: #386

### CODE REVIEW

File: packages/react-docs/src/layouts/index.js [hunk 1]
```diff
@@ -13,37 +13,43 @@ const propTypes = {
...
+  return (
+    <React.Fragment>
+      <Helmet
+        meta={[
+          { name: 'description', content: 'PatternFly React Documentation' },
+          { name: 'keywords', content: 'React, PatternFly, Red Hat' }
+        ]}
+      />
+      <Page
+        title="Patternfly React"
```
Comment (priley86) [inline packages/react-docs/src/layouts/index.js]: `PatternFly` casing on the title..

**Comment by priley86** [comment]
Was looking for `Card.test.js` but i don't see it...

**Comment by dmiller9911** [comment]
@pattrick nothing to test so far really would just be a throwaway. I will add one to verify extra props are passed through.  That should keep that from breaking.

**Comment by jgiardino** [comment]
Just wanted to mention https://github.com/patternfly/patternfly-next/issues/430 here.

Currently, the Card is implemented with `<div>` elements, but ideally, the consumer should be able to specify more semantic elements, which might vary depending on context. This issue mentions `<article>` in the description, and then there are a few more cases that I listed in a comment. I would expect the react component to allow the consumer to choose which of these variations they want.

I'm not sure if you want to go ahead and update the react component based on changes expected to the core html component, or not, but wanted to make sure you were aware of this expectation.

**Comment by dmiller9911** [comment]
I pushed up a change to allow users to pass in their own "components", which includes HTML elements. I followed the suggestion from that issue.  If we need to make additional changes later we can. We could possibly even lockdown, with warnings, which elements are allowed if we make a list.

**Comment by jeff-phillips-18** [comment]
I would be better to keep PRs limited to the intent and not include additional changes. I see changes to Title and other things that don't necessarily pertain to the card component.

**Comment by dmiller9911** [comment]
@jeff-phillips-18, while it looks like these, are unrelated...they are not.  The upgrade of patternfly-next caused all the snapshot updates and forced the title size updates.  The docs navigation is due to the fact that Card is the first component with multiple components.  I needed to filter out the secondary components.  I have closed and will piece these together as separate PR's though.

**Comment by jeff-phillips-18** [comment]
OK, gotcha @dmiller9911.  Sorry, we could have proceeded then.

**Pull request closed without merging**